### PR TITLE
feat: add home work 1

### PR DESCRIPTION
**Pull request by matveyprupas**
There is only task 1 from HW1

https://epam.sharepoint.com/sites/LMSO/Program/318459c0b61e-edc9-1d99-e518-4606666a0fdd/Scope,%20Hoisting,%20Closures.pdf

### CODE REVIEW

**Review by NikitaSpirit — COMMENTED** [review]
@matveyprupas Could you please separate tasks by files and create separate PR for each task?
You can also delete .html file

**Comment by matveyprupas** [comment]
> @matveyprupas Could you please separate tasks by files and create separate PR for each task? You can also delete .html file

Should I create a new branch for task2?

File: hw1/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function runningAverage() {
+    let arr = [];
+    let average = 0;
+    return function (arg) {
+        arr.push(arg);
+        average = (arr.reduce( (sum, e) => sum + e)) / arr.length;
```
Comment (NikitaSpirit) [inline hw1/index.js]: You run `reduce` function for every function call which is not the optimal option in terms of calculation time. Can we improve it?

Comment (matveyprupas) [reply hw1/index.js]: Yep, I've fixed it. Is it improved?

**Comment by NikitaSpirit** [comment]
@matveyprupas It would be better just to store two values:
1) sum value
2) number of times your  function was called
and you return sum / number of times function called. In my opinion it would be better solution

**Comment by matveyprupas** [comment]
@NikitaSpirit fixed it in next commit

**Review by NikitaSpirit — APPROVED** [review]
Good job!

**Pull request merged**